### PR TITLE
SV-23 View Task Details

### PR DESCRIPTION
**Pull request by mackella**
# What?

* Refactor `CrmGateway` to share common functionality across API calls.
* Add `getTask` API call to `CrmGateway`
* Refine passing of error state between CRM API and frontend

# Why?

* Complete end to end data flow

# Checklist

- [x] Unit tests
- [ ] Integration tests
- [ ] Cypress tests

# Notes

Please review the CrmGateway changes particularity.
Not sure how the CrmToken acquisition works - it seems to be misconfigured in dev - so haven't been able to test end to end.

# Next steps

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ MaTApp.getInitialProps = async (context: AppContext) => {
     context.ctx.res &&
     context.ctx.pathname !== unauthenticatedLandingPage &&
     !context.ctx.pathname.startsWith('/api')) {
+    console.log("No valid token - redirecting from " + context.ctx.pathname);
```
Comment (LBHTKarki) [inline src/pages/_app.tsx:37]: do we need this on prod?

File: src/mappings/crmToTask.ts [hunk 2]
```diff
@@ -1,55 +1,81 @@
...
+  crmTasks.value.forEach((element: any) => {
+    const task: Task = convertCrmTaskToTask(element as CrmTaskValue);
     taskArray.push(task)
   })
-  return taskArray
+
+  return taskArray;
+}
+
+function convertCrmTaskToTask(crmTask: CrmTaskValue) {
```
Comment (LBHTKarki) [inline src/mappings/crmToTask.ts:23]: This makes it easier to handle both individual and multiple tasks 👍

File: serverless.yml [hunk 3]
```diff
@@ -115,6 +115,8 @@ resources:
 
 custom:
   stage: ${opt:stage, self:provider.stage}
+  serverless-offline:
+      noPrependStageInUrl: true
```
Comment (LBHTKarki) [inline serverless.yml:119]: :thumbsup:

File: src/pages/_app.tsx (hunk 1 above)
Comment (mackella) [reply src/pages/_app.tsx:37]: Not sure... I've found it useful for debugging and it'll only run server-side. What do you think?

File: src/pages/api/tasks/[id].ts [hunk 4]
```diff
@@ -15,13 +19,13 @@ export default async (req: NextApiRequest, res: NextApiResponse<Data>) => {
...
     const getTask = new GetTask(id);
     const response = await getTask.execute();
-
-    if (response.error === undefined) {
+    
+    if (response.body) {
       res.status(200).json(response.body);
     } else {
-      res.status(response.error).json(response.body);
+      res.status(404).json({ error: 'task not found' });
```
[the remark this answers is not in the document]
Comment (mackella) [reply src/pages/api/tasks/[id].ts:26]: Agreed. I'll change it to pass through the underlying status code.

File: src/pages/_app.tsx (hunk 1 above)
Comment (LBHTKarki) [reply src/pages/_app.tsx:37]: No harm in it I suppose and helps locally

**Pull request merged**